test/test_sklearn.py

```diff
@@ -0,0 +1,74 @@
+from __future__ import annotations
+
+from typing import Callable, Type
+
+import platform
+from itertools import product
+
+from pynisher import (
+    CpuTimeoutException,
+    MemoryLimitException,
+    PynisherException,
+    WallTimeoutException,
+    contexts,
+    limit,
+    supports,
+)
+
+import pytest
+
+
+def train_svr(n_samples: int, n_features: int) -> None:
+    """Trains an svr model"""
+    from sklearn.datasets import make_regression
+    from sklearn.svm import SVR
+
+    X, y = make_regression(n_samples=n_samples, n_features=n_features)
+    SVR().fit(X, y)
+
+
+def train_svc(n_samples: int, n_features: int) -> None:
+    """Trains an svm model"""
+    from sklearn.datasets import make_classification
+    from sklearn.svm import SVC
+
+    X, y = make_classification(n_samples=n_samples, n_features=n_features)
+    SVC().fit(X, y)
+
+
+constraints_with_exceptions = [
+    (CpuTimeoutException, {"cpu_time": (1, "s")}),
+    (MemoryLimitException, {"memory": (500, "kB")}),
+    (WallTimeoutException, {"wall_time": (1, "s")}),
```

Comment:
Do we know if this exception is triggered in the SVM code or rather in the dataset construction code?

Reply:
The only way I can think to check that is `try catch` around everything that isn't `fit`

Reply:
Would that make sense? Limiting scikit-learn's memory and time usage is the main use case of the pynisher, so it would be good if we test all kinds of edge cases here.

Reply:
Yup, I agree, I'll try to more testing and ping you with more info.